pandas/core/indexes/multi.py

```diff
@@ -3095,8 +3105,56 @@ def _update_indexer(idxr, indexer=indexer):
         # empty indexer
         if indexer is None:
             return Int64Index([])._ndarray_values
+
+        if need_sort:
+            indexer = self._reorder_indexer(seq, indexer)
+
         return indexer._ndarray_values
 
+    def _reorder_indexer(self, seq, indexer: Int64Index) -> Int64Index:
+        """
+        Reorder an indexer of a MultiIndex (self) so that the label are in the
+        same order as given in seq
+
+        Parameters
+        ----------
+        seq : label/slice/list/mask or a sequence of such
+        indexer: an Int64Index indexer of self
+
+        Returns
+        -------
+        indexer : a sorted Int64Index indexer of self ordered as seq
+        """
+        from typing import Tuple
+
+        n = len(self)
+        keys = tuple()  # type: Tuple[np.ndarray, ...]
```

Comment:
you can use the py36 form here

Reply:
Do you mean no type hint?
I add this when mypy complained with the following
`pandas/core/indexes/multi.py:3130: error: Need type annotation for 'keys'`